src/RestClient.pas

```diff
@@ -324,17 +327,19 @@ function TRestClient.DoRequest(Method: TRequestMethod; ResourceRequest: TResourc
 
     if assigned(OnBeforeRequest) then
       OnBeforeRequest(self, ResourceRequest, Method);
+
+    ResourceRequest.GetContent.Position := 0;
```

Comment:
if you read the Content in the `OnBeforeRequest` Event it changes the position and messes up the request, so I think it is safer to reset it here again.